app/workers/commit_monitor_handlers/commit_range/hamllint_checker.rb

```diff
@@ -0,0 +1,109 @@
+class CommitMonitorHandlers::CommitRange::HamllintChecker
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  def self.handled_branch_modes
+    [:pr]
+  end
+
+  attr_reader :branch, :commits, :results, :github
+
+  def perform(branch_id, new_commits)
+    @branch  = CommitMonitorBranch.where(:id => branch_id).first
+
+    if @branch.nil?
+      logger.info("Branch #{branch_id} no longer exists.  Skipping.")
+      return
+    end
+    unless @branch.pull_request?
+      logger.info("Branch #{@branch.name} is not a pull request.  Skipping.")
+      return
+    end
+
+    @commits = @branch.commits_list
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    diff_details = filter_ruby_files(diff_details_for_branch)
+    files        = diff_details.keys
+
+    if files.length == 0
+      @results = {"files" => []}
+    else
+      @results = rubocop_results(files)
+      @results = RubocopResultsFilter.new(@results, diff_details).filtered
+    end
+
+    write_to_github
```

Comment:
I think this checker should work slightly differently that the rubocop one in that it shouldn't write any comment at all if there were no haml files in the PR.  What do you think?